.github/workflows/pypi-publish.yaml

```diff
@@ -1,35 +1,71 @@
 name: Publish Python Package
+# Upload to PyPI is triggered by creating & publishing a release in GitHub UI
 
 on:
   release:
+    # Trigger the workflow only for real releases but no for draft releases
     types: [created]
 
 jobs:
-  build-n-publish:
-    name: Build and publish Python 🐍 distributions 📦 to PyPI
+  build:
+    name: Build Python 🐍 distributions 📦 for publishing to PyPI
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4.2
+      # By default the ref (branch, tag or SHA) that triggered the workflow,
+      # is checked out. So the version tag is available for dynamic versioning
+      # and the option "with: fetch-depth: 0" is not needed here.
 
     - name: Set up Python
-      uses: actions/setup-python@v2.2.2
+      uses: actions/setup-python@v5.5
       with:
         python-version: 3.12
 
     - name: Install Poetry
-      run: pipx install poetry
-
-    - name: Install dependencies
-      run: poetry install --no-interaction
+      run: |
+        pipx install poetry
+        pipx inject poetry poetry-dynamic-versioning
 
     - name: Build source and wheel archives
       run: |
-        poetry version $(git describe --tags --abbrev=0)
         poetry build
+        # Check if correct tag is available
+        git describe --tags --abbrev=0
```

Comment:
what happens if the correct tag isn't available?

Reply:
Hmm. I just found out that it is surprisingly possible to create a release without selecting a tag on the release-creation page. So it may be safer to not rely on it and use the full checkout with tags to derive the version safely. I'll change this.